dftimewolf/lib/telemetry.py

```diff
@@ -0,0 +1,150 @@
+"""Telemetry module."""
+import datetime
+from dataclasses import dataclass
+from typing import Dict, Any, List, Union
+import uuid
+
+# mypy complains when doing from google.cloud import spanner
+try:
+  from google.cloud import spanner  # type: ignore
+  HAS_SPANNER = True
+except ImportError:
+  HAS_SPANNER = False
+
+from dftimewolf import config
+
+
+@dataclass
+class TelemetryEntry:
+  """A simple dataclass to store module-related statistics.
+
+  Attributes:
+    module_type: Type of the module that generated the telemetry.
+    module_name: Name of the module that generated the telemetry. This has the
+        same value as module_type when no runtime_name has been specified for
+        the module.
+    telemetry: Dictionary of telemetry to store. Contents are arbitrary, but
+        keys must be strings.
+  """
+  module_type: str
+  module_name: str
+  telemetry: Dict[str, str]
```

Comment:
You mention keys must be strings in the docstring, but type the value as `str` as well. Should the type be `dict[str, Any]`?

Reply:
Done.